app-text/wordgrinder/wordgrinder-0.7.1.recipe

```diff
@@ -11,40 +11,52 @@ REVISION="1"
 SOURCE_URI="https://github.com/davidgiven/wordgrinder/archive/$portVersion.tar.gz"
 CHECKSUM_SHA256="54085af7963e1f67342bc0b1b20d1ccc75494f2e23d401b601cf3089acea747c"
 
-ARCHITECTURES="!x86_gcc2 !x86_64"
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
 
 PROVIDES="
-	wordgrinder = $portVersion
-	cmd:WordGrinder = $portVersion
+	wordgrinder$secondaryArchSuffix = $portVersion
+	cmd:wordgrinder$commandSuffix = $portVersion
 	"
 REQUIRES="
-	haiku
-	lib:libncurses
-	lib:liblua
-	lib:libz
+	haiku$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.3
+	lib:libz$secondaryArchSuffix
 	"
 
 BUILD_REQUIRES="
-	haiku_devel
-	devel:libncurses
-	devel:liblua
-	devel:libz
+	haiku${secondaryArchSuffix}_devel
+	devel:libncurses$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.3
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:g++
+	cmd:cc$secondaryArchSuffix
 	cmd:make
 	cmd:ninja
-	cmd:pkg_config
+	cmd:pkg_config$secondaryArchSuffix
 	"
 
 BUILD()
 {
-	make OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
-		DOCDIR=$docDir MANDIR=$manDir
+	make PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
+		MANDIR=$manDir HOME=$prefix LUA_PACKAGE=lua
```

Comment:
Hmm, my previous suggestion to pass HOME=$prefix might be risky.
We should probably set HOME to something better. What about `finddir B_USER_DIRECTORY` ?
```diff
 BUILD()
 {
+	HOME="`finddir B_USER_DIRECTORY`"
	make PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
-		MANDIR=$manDir HOME=$prefix LUA_PACKAGE=lua
+		MANDIR=$manDir HOME="$HOME" LUA_PACKAGE=lua
 }
 
 INSTALL()
 {
+	HOME="`finddir B_USER_DIRECTORY`"
 	make install PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
-		MANDIR=$manDir HOME=$prefix LUA_PACKAGE=lua
+		MANDIR=$manDir HOME="$HOME" LUA_PACKAGE=lua
```

Reply:
Changes comming up